src/autolua.zig

```diff
@@ -77,6 +78,20 @@ pub fn push(L: ?*lua.lua_State, value: anytype) void {
                 // TODO: fill in the metatable with info about the type?
             }
         },
+        .Array => |A| {
+            lua.lua_createtable(L, 0, A.len);
+            for (value) |x, i| {
+                push(L, x);
+                lua.lua_rawseti(L, -2, @intCast(c_int, i+1));
+            }
+        },
+        .Optional => |N| {
+            if (value == null) {
+                lua.lua_pushnil(L);
+                return;
```

Comment:
normally I would say this should be `return push(L, null);` but given the above comment we may need to figure it out.